### PR TITLE
Fixes #195: Merge operation statuses [WIP]

### LINKED ISSUE #195: Merge operation statuses
There are 3 different-but-similar statuses `ClaimingStatus`, `ExecutionStatus` and `TxSendErrors`, the only difference is the error text but the code that handles is pretty much the same:

https://github.com/ethereum-alarm-clock/timenode-core/blob/318a0e866154dd3b01035ab218d2f0cee65ff3c0/src/Actions/Actions.ts#L91-L98

https://github.com/ethereum-alarm-clock/timenode-core/blob/318a0e866154dd3b01035ab218d2f0cee65ff3c0/src/Actions/Actions.ts#L43-L50

https://github.com/ethereum-alarm-clock/timenode-core/blob/318a0e866154dd3b01035ab218d2f0cee65ff3c0/src/Enum/TxSendErrors.ts#L1-L8

https://github.com/ethereum-alarm-clock/timenode-core/blob/318a0e866154dd3b01035ab218d2f0cee65ff3c0/src/Enum/ExecuteStatus.ts#L1-L7

Expected:

- [ ] Remove `ClaimingStatus`, `ExecutionStatus` and `TxSendErrors` and use unified transaction status.
- [ ] Remove create context specific logs to keep verbose logging

### PR DESCRIPTION
**Pull request by anshumanv**
Fixes #195

### CODE REVIEW

**Comment by anshumanv** [comment]
@kosecki123 added separate pending and failed statuses for claim, execute.

Please see if this looks okay.

**Comment by anshumanv** [comment]
ping @kosecki123 :sweat_smile:

**Comment by Kuzirashi** [comment]
Hello @anshumanv, after the changes integration tests are failing. Please see Travis CI checks:
https://travis-ci.org/ethereum-alarm-clock/timenode-core/jobs/421134065

**Comment by anshumanv** [comment]
Ah sorry, looking it up. :+1:

**Comment by anshumanv** [comment]
It seems to be failing in all the PRs, not sure why. :thinking:

**Comment by Kuzirashi** [comment]
Hey @anshumanv, it is passing on master. I just restarted build for you to confirm it's not a problem with tests. If fails on bignumber and ts-node PR since they break application and this is expected they fail, please don't use them as an indicator. 

Failing test is execution test:
```
 1) TimeNode
       executes transaction:
     Error: Timeout of 400000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/travis/build/ethereum-alarm-clock/timenode-core/test/e2e/TestTimeNode.ts)
```

And, proof that tests work:

- Master: https://travis-ci.org/ethereum-alarm-clock/timenode-core/builds/421065710 (passing)
- This branch: https://travis-ci.org/ethereum-alarm-clock/timenode-core/builds/421134063

Just a little bit of fixing and we can review your work! :blush:

File: src/Enum/TxSendErrors.ts [hunk 1]
```diff
@@ -1,8 +0,0 @@
-export enum TxSendErrors {
```
Comment (kosecki123) [inline src/Enum/TxSendErrors.ts]: There are good error enum, but the descriptions are not really descriptive.

File: src/Enum/Status.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  OK = 'OK',
+  CLAIM_FAILED = 'Claiming: Transaction already claimed',
+  EXECUTE_FAILED = 'Execution: Unable to send the execute action',
+  TX_FAILED = 'FAILED',
+  CLAIM_PENDING = 'Claiming: Skipped - Other claiming found',
+  EXECUTE_PENDING = 'Execution: Skipped - Other execution found',
+  NOT_ENABLED = 'Claiming: Skipped - Claiming disabled',
+  ACCOUNT_BUSY = 'Claiming: Skipped - Account is busy',
+  SUCCESS = 'Success'
+}
```
Comment (kosecki123) [inline src/Enum/Status.ts]: This enum contains a lot of noise, please stick to previous enum definition. Maybe decouple message from enum or use enum string as templates. I don't really see a need to keep both ACCOUNT_BUSY and WALLET_BUSY and other duplicated ones.

File: src/Actions/Actions.ts [hunk 3]
```diff
@@ -87,24 +86,24 @@ export default class Actions implements IActions {
...
       switch (status) {
-        case TxSendErrors.OK:
+        case Status.OK:
           return await this.accountExecution(txRequest, receipt, opts, from);
-        case TxSendErrors.WALLET_BUSY:
-          return ExecuteStatus.WALLET_BUSY;
-        case TxSendErrors.IN_PROGRESS:
-          return ExecuteStatus.IN_PROGRESS;
+        case Status.WALLET_BUSY:
+          return Status.WALLET_BUSY;
```
Comment (kosecki123) [inline src/Actions/Actions.ts]: To be removed. And other similar cases in the code.

File: src/Enum/TxSendErrors.ts (hunk 1 above)
Comment (anshumanv) [reply src/Enum/TxSendErrors.ts]: This file is removed, can you please clarify what you're referring to? :thinking:

File: src/Enum/Status.ts (hunk 2 above)
Comment (anshumanv) [reply src/Enum/Status.ts]: Okay will keep a single status like before, no problem, but for almost similar statuses like `FAILED` for `EXECUTE` and `CLAIM` how do I manage the differences, do I use ``Claiming: ${Status.FAILED}`` for claim fail and ``Execution ${Status.FAILED}`` this way we can have single `FAILED` status in the file, I'm sorry if I don't understand it very clearly. @kosecki123

**Comment by lsaether** [comment]
@anshumanv You need to have a local instance of `ganache-cli` running for the tests to run successfully on a local machine. You can look at the Travis configuration in `.travis.yml` to see how to set up the environment. 

Basically, you'll want to open one screen and run `npm run ganache` then in another screen run `npm run deploy-eac` and after completing these two steps you should be able to run `npm run test` and the tests will run correctly.

**Pull request closed without merging**